### PR TITLE
[RLlib] Fix MB MPO

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

https://github.com/ray-project/ray/issues/38321 raises the issue that the MB MPO example does not work.
This PR fixes multiple issues within MBMPO that have gone undedected in the past:

- When initializing the loss, we use a batch size of 32, which is incompatible with the MAML loss function.
- Since gymnasium has changed with time, the bare Pendulum and CartPole envs that we are wrapping don't have any logic that truncates or terminates by default. Only when you create them with gym.make(...), wrappers are added accordingly. Therefore, our old technique of wrapping causes MBMPO to collect endlessly initially.
- Later, when MBMPO uses a dynamics model to predict future observations, the dynamics model also lacks a truncation or termination mechanism. This PR also adds this to the envs. I'm not sure how this was done in the past.

### CODE REVIEW

**Review by sven1977 — APPROVED** [review]
Nice fix! Thanks @ArturNiederfahrenhorst for digging into this.

**Comment by MrSVF** [comment]
Dear @ArturNiederfahrenhorst, thank you for your pull request!  The mbmpo algorithm runs correctly now, but unfortunately it does not train properly. Could you analyze this issue https://github.com/ray-project/ray/issues/40400 and give your recommendation, why this is happening and how to correct this behavior?
